### PR TITLE
fix(user): change user to www-data in config files

### LINKED ISSUE #10: `chown www-data` but httpd runs as user `daemon`
The default httpd.conf sets `User daemon`.  So it's kind of odd to have `chown www-data` in the Dockerfile.  (www-data is the default when you install the apache packages from debian, as opposed to the upstream tarball).

It's not exactly a bug - it works because the directory is readable by any user.  However it could cause confusion if you need Apache to have write access somewhere.

Fortunately the PHP+Apache image sets `User www-data` in httpd.conf.  Maybe `User www-data` (and Group) should be added to the `sed` command in this Dockerfile, which edits the config file?

### PR DESCRIPTION
**Pull request by fzipi**
Signed-off-by: Felipe Zipitria <felipe.zipitria@owasp.org>

Fixes #10.

### CODE REVIEW

**Comment by fzipi** [comment]
@tianon What do you think about this one?

File: 2.4/Dockerfile [hunk 1]
```diff
@@ -203,6 +203,8 @@ RUN set -eux; \
 		-e 's!^(\s*CustomLog)\s+\S+!\1 /proc/self/fd/1!g' \
 		-e 's!^(\s*ErrorLog)\s+\S+!\1 /proc/self/fd/2!g' \
 		-e 's!^(\s*TransferLog)\s+\S+!\1 /proc/self/fd/1!g' \
+		-e 's!^User daemon!User www-data!g' \
+		-e 's!^Group daemon!Group www-data!g' \
 		"$HTTPD_PREFIX/conf/httpd.conf" \
 		"$HTTPD_PREFIX/conf/extra/httpd-ssl.conf" \
 	; \
```
Comment (tianon) [inline 2.4/Dockerfile:210]: This seems like an OK solution, thanks.

However, I think we should both match the format of the existing regular expressions, and add a check that we did update them successfully/correctly (so we don't accidentally regress in the future), something like:

```suggestion
		-e 's!^(\s*User)\s+daemon\s*$!\1 www-data!g' \
		-e 's!^(\s*Group)\s+daemon\s*$!\1 www-data!g' \
		"$HTTPD_PREFIX/conf/httpd.conf" \
		"$HTTPD_PREFIX/conf/extra/httpd-ssl.conf" \
	; \
	grep -E '^\s*User www-data$' "$HTTPD_PREFIX/conf/httpd.conf"; \
	grep -E '^\s*Group www-data$' "$HTTPD_PREFIX/conf/httpd.conf"; \
```

**Comment by fzipi** [comment]
@tianon Thanks. Looks good. Propagated the suggestion to alpine also.

File: 2.4/alpine/Dockerfile [hunk 2]
```diff
@@ -203,9 +203,13 @@ RUN set -eux; \
 		-e 's!^(\s*CustomLog)\s+\S+!\1 /proc/self/fd/1!g' \
 		-e 's!^(\s*ErrorLog)\s+\S+!\1 /proc/self/fd/2!g' \
 		-e 's!^(\s*TransferLog)\s+\S+!\1 /proc/self/fd/1!g' \
+    	-e 's!^(\s*User)\s+daemon\s*$!\1 www-data!g' \
+		-e 's!^(\s*Group)\s+daemon\s*$!\1 www-data!g' \
```
Comment (tianon) [inline 2.4/alpine/Dockerfile:207]: Whitespace is a bit off here :eyes:

Comment (fzipi360) [reply 2.4/alpine/Dockerfile:207]: Fixed!

Comment (fzipi) [reply 2.4/alpine/Dockerfile:207]: Should be fixed.

Comment (fzipi) [reply 2.4/alpine/Dockerfile:207]: Sorry, you were right. It wasn't 🤦 . Now it should :)

**Review by tianon — APPROVED** [review]

**Pull request merged**